server/datastore/mysql/apple_mdm_test.go

```diff
@@ -1821,20 +1817,18 @@ func testMDMAppleInsertIdPAccount(t *testing.T, ds *Datastore) {
 	err = ds.InsertMDMIdPAccount(ctx, acc)
 	require.NoError(t, err)
 
-	// try to insert an empty account
-	err = ds.InsertMDMIdPAccount(ctx, &fleet.MDMIdPAccount{})
-	require.Error(t, err)
-
 	// duplicated values get updated
-	acc.SaltedSHA512PBKDF2Dictionary.Iterations = 3000
-	acc.SaltedSHA512PBKDF2Dictionary.Salt = []byte("tlas")
-	acc.SaltedSHA512PBKDF2Dictionary.Entropy = []byte("yportne")
 	err = ds.InsertMDMIdPAccount(ctx, acc)
 	require.NoError(t, err)
-	var out fleet.MDMIdPAccount
-	err = sqlx.GetContext(ctx, ds.reader, &out, "SELECT * FROM mdm_idp_accounts WHERE uuid = 'ABC-DEF'")
+
```

Comment:
I can't comment on unmodified lines above, but with the deleted changes to `acc` before the previous call to `InsertMDMIdPAccount`, it's not a `duplicated values get updated` test anymore, it's just the same as the `try to instert the same account` one above. I think you should either remove it, or change e.g. the full name before the call to check that it did get updated.

Reply:
I wanted to ping you and @gillespi314 to ask about this in general and I forgot. Should we enforce an uniqueness constraint on username? or username+fullname? 

with the current logic, an user unboxing a machine for the second time will have 2 tokens, which I think it's ok but I don't know what do you think?